R/createReport.R

```diff
@@ -97,116 +96,20 @@ createReport = function(txt_folder = NULL, mztab_file = NULL, yaml_obj = list(),
   ##
-  ## determines if a local mqpar.xml should be used to grep all YAML parameters whose name starts with "MQpar_" from the
-  ## original mqpar.xml instead of the yaml.config. The "MQpar_..." param from the config
-  ## will be ignored and the newly written yaml.config will contain the values from mqpar.xml.
-  param_useMQPAR = yc$getYAML("PTXQC$UseLocalMQPar", TRUE)
-  
-  add_fs_col = yc$getYAML("PTXQC$NameLengthMax_num", 14)
-  
-  id_rate_bad = yc$getYAML("File$Summary$IDRate$Thresh_bad_num", 20, 0, 100)
-  id_rate_great = yc$getYAML("File$Summary$IDRate$Thresh_great_num", 35, 0, 100)
-  
-  GL_name_min_length = 8
-  
-  pg_ratioLabIncThresh = yc$getYAML("File$ProteinGroups$RatioPlot$LabelIncThresh_num", 4)
-  ## default median intensity in log2 scale
-  param_PG_intThresh = yc$getYAML("File$ProteinGroups$IntensityThreshLog2_num", 25, 1, 100)
-
-  ## get scoring threshold (upper limit)
-  param_EV_protThresh = yc$getYAML("File$Evidence$ProteinCountThresh_num", 3500, 1, 1e5)
-
-  ## default median intensity in log2 scale
-  param_EV_intThresh = yc$getYAML("File$Evidence$IntensityThreshLog2_num", 23, 1, 100)
-
-  ## get scoring threshold (upper limit)
-  param_EV_pepThresh = yc$getYAML("File$Evidence$PeptideCountThresh_num", 15000, 1, 1e6)
 
-  ### warn of special contaminants!
-  ## these need to be in FASTA headers (description is not enough)!
-  ## syntax:  list( contaminant1 = c(name, threshold), contaminant2 = c(...), ...)
-  ##
-  ##  if within the YAML file
-  ##    SpecialContaminants: no
-  ##  is set, then 'yaml_contaminants' will be 'FALSE'
-  ##
-  contaminant_default = list("cont_MYCO" = c(name="MYCOPLASMA", threshold=1)) # name (FASTA), threshold for % of unique peptides
-  ##contaminant_default = FALSE ## to switch it off by default
-  yaml_contaminants = yc$getYAML("File$Evidence$SpecialContaminants", contaminant_default)
-  
-  param_EV_MatchingTolerance = yc$getYAML("File$Evidence$MQpar_MatchingTimeWindow_num", 1)
-  if (param_useMQPAR &! MZTAB_MODE) {
-    v = getMQPARValue(txt_files$mqpar, "matchingTimeWindow") ## will also warn() if file is missing
-    if (!is.null(v)) {
-      param_EV_MatchingTolerance = yc$setYAML("File$Evidence$MQpar_MatchingTimeWindow_num", as.numeric(v))
-    }
-  }
-  param_evd_mbr = yc$getYAML("File$Evidence$MatchBetweenRuns_wA", "auto")
+  param_lstqcMetrics_list <- createYaml(yc = yc, path = rprt_fns$yaml_file, DEBUG_PTXQC = DEBUG_PTXQC, txt_files = txt_files)
+  param <- param_lstqcMetrics_list[[1]]
+  lst_qcMetrics_ord <- param_lstqcMetrics_list[[2]]
   
-  param_EV_PrecursorTolPPM = yc$getYAML("File$Evidence$MQpar_firstSearchTol_num", 20)
-  if (param_useMQPAR & !MZTAB_MODE) {
-    v = getMQPARValue(txt_files$mqpar, "firstSearchTol") ## will also warn() if file is missing
-    if (!is.null(v)) {
-      param_EV_PrecursorTolPPM = yc$setYAML("File$Evidence$MQpar_firstSearchTol_num", as.numeric(v))
-    }
-  }
+  ##not sure if it updates automatically
```

Comment:
is `lst_qcMetrics_ord` needed afterwards?
Why not simply assign to `lst_qcMetrics <- param_lstqcMetrics_list$metrics` and be done?

Reply:
lst_qcMetrics_ord is used one time afterwards in line 656, could probably be replaced by lst_qcMetrics?
`for (qcm in lst_qcMetrics_ord){...}`